src/main/java/com/gmail/goosius/siegewar/utils/SiegeWarSicknessUtil.java

```diff
@@ -0,0 +1,156 @@
+package com.gmail.goosius.siegewar.utils;
+
+import com.gmail.goosius.siegewar.SiegeController;
+import com.gmail.goosius.siegewar.enums.SiegeWarPermissionNodes;
+import com.gmail.goosius.siegewar.objects.Siege;
+import com.gmail.goosius.siegewar.settings.SiegeWarSettings;
+import com.gmail.goosius.siegewar.settings.Translation;
+import com.palmergames.bukkit.towny.Towny;
+import com.palmergames.bukkit.towny.TownyAPI;
+import com.palmergames.bukkit.towny.TownySettings;
+import com.palmergames.bukkit.towny.TownyUniverse;
+import com.palmergames.bukkit.towny.exceptions.NotRegisteredException;
+import com.palmergames.bukkit.towny.object.Nation;
+import com.palmergames.bukkit.towny.object.Resident;
+import com.palmergames.bukkit.towny.object.Town;
+import com.palmergames.bukkit.util.BukkitTools;
+import com.palmergames.util.TimeTools;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SiegeWarSicknessUtil {
+
+    public static void punishNonSiegeParticipantsInSiegeZone() {
+
+        for (Player player : BukkitTools.getOnlinePlayers()) {
+            Location location = player.getLocation();
+            List<Siege> sieges = SiegeController.getSiegesAt(location);
+            Resident resident = TownyUniverse.getInstance().getResident(player.getUniqueId());
+            TownyUniverse universe = TownyUniverse.getInstance();
+
+            // not in a siege zone
+            if (sieges.isEmpty())
+                continue;
+
+            // Players immune to war nausea won't be punished for this
+            if (universe.getPermissionSource().testPermission(player,
+                    SiegeWarPermissionNodes.SIEGEWAR_IMMUNE_TO_WAR_NAUSEA.getNode()))
+                continue;
```

Comment:
This could be flipped around to rule out false positives earlier on:
```java
            // Players immune to war nausea won't be punished for this
            if (player.has(SiegeWarPermissionNodes.SIEGEWAR_IMMUNE_TO_WAR_NAUSEA.getNode()))
                continue;

            List<Siege> sieges = SiegeController.getSiegesAt(player.getLocation());
            // not in a siege zone
            if (sieges.isEmpty())
                continue;

            Resident resident = TownyUniverse.getInstance().getResident(player.getUniqueId());
            if (resident == null) 
                continue;
```